### PR TITLE
SFTP: add strategy  in the deployment

### PR DESCRIPTION
**Pull request by romainjacquet**
This is a simple patch to add support of strategy in the deployment.
Tests and values has been updated according to the changes.
I added this feature because with some cloud providers, like OVH, there is no support PVC with ReadWriteMany. So without strategy like Recreate it do not work.

### CODE REVIEW

File: charts/sftp-server/values.yaml [hunk 1]
```diff
@@ -156,6 +156,12 @@ probes:
   # -- Enable readiness probe
   readiness: true
 
+#
+# ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
```
Comment (sj14) [inline charts/sftp-server/values.yaml:160]: Please use the `# -- ` prefix, so the comment will be part of the generated readme.
Please also wrap the ref into a markdown link as done in other comments (`([doc](<URL>`).

**Review by sj14 — COMMENTED** [review]
Thanks for your contribution 🎉 
I have two nitpicks, otherwiese LGTM.

File: charts/sftp-server/README.md [hunk 2]
```diff
@@ -72,3 +72,4 @@ helm upgrade sftp --install sj14/sftp-server
 | sftp.users[0].pubKeys | list | `[]` | public user keys ([doc](https://github.com/atmoz/sftp/blob/ffeb104beec76cc622abda34ee2132c790b5559c/README.md#logging-in-with-ssh-keys)) |
 | sftp.users[0].uid | string | `""` |  |
 | tolerations | list | `[]` |  |
+| updateStrategy.type | string | `"RollingUpdate"` |  |
```
[the remark this answers is not in the document]
Comment (romainjacquet) [reply charts/sftp-server/README.md:75]: You are right, this is not good. As you can also specify parameters like rollingUpdate.maxSurge.

File: charts/sftp-server/values.yaml (hunk 1 above)
Comment (romainjacquet) [reply charts/sftp-server/values.yaml:160]: Ok you are right

**Comment by sj14** [comment]
I've removed the `rollingUpdate` section from the values.yaml file, because it was empty, so I don't think it was useful (https://github.com/sj14/helm-charts/pull/69/commits/cd9797cd59631fb5bed8d2348a8c8baf14c38294)

Otherwise, thanks again for your contribution 🎉

**Pull request merged**